src/codemagic/tools/app_store_connect.py

```diff
@@ -492,25 +492,53 @@ def list_devices(
         "register-device",
         BundleIdArgument.PLATFORM,
         DeviceArgument.DEVICE_NAME,
-        DeviceArgument.DEVICE_UDID,
+        DeviceArgument.DEVICE_UDIDS,
+        DeviceArgument.IGNORE_REGISTRATION_ERRORS,
     )
     def register_device(
         self,
         platform: BundleIdPlatform,
         device_name: str,
-        device_udid: str,
```

Comment:
Please retain the old `device_udid: str` positional argument for backwards compatibility as optional on the same position.
```python
device_udid: Optional[str] = None,
```
We don't want to introduce breaking changes to Python API. Final UDIDs can be composed by combining new UDIDs args and the legacy one. If no UDIDs are given, then fail with error.